lib/assessment.js

```diff
@@ -4,23 +4,33 @@
 // Write a function that takes zero or more arguments
 // and returns the minimum from that set
 // assume that all the argument are numbers
-var min = function min() {
-
+var min = function min(nums) {
+  return Math.min(nums)
 };
 
+
 // Question 2
 // Write a function that takes zero or more arguments
 // and returns the sum of that set
 // assume that all the argument are numbers
-var sum = function sum() {
-
+var sum = function sum(addends) {
+  var adding = 0;
+  for (i = 0; i < addends.length; i++) {
+    adding += addends[i];
+  }
+  return adding;
 };
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
 var newArray = function newArray(length, defaultValue) {
-
+  ary = [];
+  ary.length = length;
```

Comment:
This line will not do what you think it does. In fact, don't ever write this, because it will really mess up the array that you do it to. `.length` is a attribute of an array, not a method; the way that it's kept accurate is that every time an element gets added or removed, the `.length` value gets updated automatically to account for the change. If you manually set `.length` to a different value - say 10 - then if you add one more element, `.length` will be equal to 11 _irrespective of how long the array actually is_. See why that might be dangerous?